cloudant-client/src/main/java/com/cloudant/client/api/ClientBuilder.java

```diff
@@ -672,51 +677,93 @@ public static ClientBuilder bluemix(String vcapServices) {
      * <ul>
      *     <li>The {@code vcapServices} is {@code null}.</li>
      *     <li>The {@code vcapServices} is not valid.</li>
+     *     <li>A service with the name matching "cloudantNoSQLDB" could not be found in
+     *     {@code vcapServices}.</li>
      *     <li>An instance with a name matching {@code instanceName} could not be found in
-     *     {@code vcapServices}.
+     *     {@code vcapServices}.</li>
      *     <li>The {@code instanceName} is {@code null} and multiple Cloudant service instances
      *     exist in {@code vcapServices}.</li>
      * </ul>
      */
     public static ClientBuilder bluemix(String vcapServices, String instanceName) {
-        Gson gson = new GsonBuilder().create();
+        return bluemix(vcapServices, "cloudantNoSQLDB", instanceName);
+    }
 
+    /**
+     * Sets Cloudant client credentials by inspecting a service information JSON string. This object
+     * takes the form of a {@code VCAP_SERVICES} environment variable which is a JSON object that
+     * contains information that you can use to interact with a service instance in Bluemix.
+     * <P>
+     * Note: Specifying an instance name is only required when multiple Cloudant service instances
+     * are present. If there is only a single Cloudant service instance then
+     * {@link #bluemix(String)} can also be used.
+     * </P>
+     *
+     * @param vcapServices service information JSON string, for example the contents of
+     *                     {@code VCAP_SERVICES} environment variable
+     * @param serviceName name of Bluemix service to use
+     * @param instanceName name of Bluemix service instance or {@code null} to try to use the only
+     *                    available Cloudant service
+     * @return a new ClientBuilder for the account
+     *
+     * @throws IllegalArgumentException if any of the following conditions are true:
+     * <ul>
+     *     <li>The {@code vcapServices} is {@code null}.</li>
+     *     <li>The {@code vcapServices} is not valid.</li>
+     *     <li>A service with the name matching {@code serviceName} could not be found in
+     *     {@code vcapServices}.</li>
+     *     <li>An instance with a name matching {@code instanceName} could not be found in
+     *     {@code vcapServices}.</li>
+     *     <li>The {@code instanceName} is {@code null} and multiple Cloudant service instances
+     *     exist in {@code vcapServices}.</li>
+     * </ul>
+     */
+    public static ClientBuilder bluemix(String vcapServices, String serviceName, String instanceName) {
         if (vcapServices == null) {
             throw new IllegalArgumentException("The vcapServices JSON information was null.");
         }
 
-        CloudFoundryServices services;
+        JsonObject obj;
```

Comment:
Do we also need a `null` check for `serviceName` (and an extra bullet point in the `@throws` documentation)?

Reply:
I'll add it in. The additional null check is belt and braces really. I guess it'll save us from parsing the json and then throwing.